environs/bootstrap/bootstrap.go

```diff
@@ -218,8 +206,10 @@ func Bootstrap(ctx environs.BootstrapContext, environ environs.Environ, args Boo
 	// If no arch is specified as a constraint, we'll bootstrap
 	// on the same arch as the client used to bootstrap.
 	var bootstrapArch string
-	if bootstrapConstraints.Arch != nil {
-		bootstrapArch = *bootstrapConstraints.Arch
+	if args.BootstrapConstraints.Arch != nil {
+		bootstrapArch = *args.BootstrapConstraints.Arch
+	} else if args.ModelConstraints.Arch != nil {
```

Comment:
they do :)
if bootstrap arch is set, use it.. else if model arch is set, use it.. else use host arch..
No?